### PR TITLE
feat: add slack source

### PR DESCRIPTION
**Pull request by gitcommitshow**
## What are the changes introduced in this PR?

Adds Slack source transform

Research doc - https://www.notion.so/rudderstacks/Slack-Source-integration-119cd4bd3d9c466fb875c10dc77fc719?pvs=4

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

No

### Any new dependencies introduced with this change?

No

### Any new generic utility introduced or modified. Please explain the changes.

No

### Any technical or performance related pointers to consider with the change?

N

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3148/test-report.html

**Review by koladilip — APPROVED** [review]

**Review by yashasvibajpai — DISMISSED** [review]
Are we adding tests in this PR?

**Comment by yashasvibajpai** [comment]
@shrouti1507 added you too for a review if time permits

**Comment by gitcommitshow** [comment]
> Are we adding tests in this PR?

I wrote some tests on local but facing difficulties in verifying those on my local machine. Need suggestions on how to move forward.

File: src/v0/sources/slack/util.js
Comment (krishna2020) [inline src/v0/sources/slack/util.js:1]: @gitcommitshow you can plan for writing unit tests of `util.js` & slack integration component tests at https://github.com/rudderlabs/rudder-transformer/tree/develop/test/integrations/sources

Comment (gitcommitshow) [reply src/v0/sources/slack/util.js:1]: Thank you. I had written input/output under `https://github.com/rudderlabs/rudder-transformer/tree/develop/test/__tests__/data`, is that recommended?

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

File: src/v0/sources/slack/util.js
Comment (gitcommitshow) [reply src/v0/sources/slack/util.js:1]: I have added the tests under `/test/integrations/sources/slack/data.ts` as this is where we are organising the newer integrations. I also have the tests under older structure `/test/__tests__/data` but I am skipping those to add to this PR. If this looks correct approach, please review and merge if everything is good.

File: test/integrations/sources/slack/data.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            event_time: 1709441309,
+            token: 'REm276ggfh72Lq',
+            team_id: 'T0GFJL5J7',
+            context_team_id: 'T0GFJL5J7',
+            context_enterprise_id: null,
+            api_app_id: 'B02SJMHRR',
+            authorizations: [
+              {
+                enterprise_id: null,
+                team_id: 'T0GFJL5J7',
```
Comment (yashasvibajpai) [inline test/integrations/sources/slack/data.ts:66]: Just wanted to check, if these are actual valid creds, we can use dummy ones.

**Review by yashasvibajpai — APPROVED** [review]

File: test/integrations/sources/slack/data.ts (hunk 1 above)
Comment (gitcommitshow) [reply test/integrations/sources/slack/data.ts:66]: they are dummy credentials

File: src/v0/sources/slack/util.js
Comment (krishna2020) [reply src/v0/sources/slack/util.js:1]: I was mentioning about unit tests for testing function of `util.js`
Eg: https://github.com/rudderlabs/rudder-transformer/blob/develop/src/v0/destinations/mp/util.test.js

Comment (krishna2020) [reply src/v0/sources/slack/util.js:1]: `/test/integrations/sources/slack/data.ts` --> yes, this is the right one for component tests

File: test/integrations/sources/slack/data.ts
Comment (krishna2020) [inline test/integrations/sources/slack/data.ts:1]: Add one scenario for generating track event

File: src/v0/sources/slack/util.js
Comment (gitcommitshow) [reply src/v0/sources/slack/util.js:1]: thank you for the feedback. added the unit tests.

File: test/integrations/sources/slack/data.ts
Comment (gitcommitshow) [reply test/integrations/sources/slack/data.ts:1]: done. added one track event ("Message") test

**Review by krishna2020 — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Comment by gitcommitshow** [comment]
Tested on prod. This is working fine.

**Pull request merged**